components/core/src/clp/Grep.cpp

```diff
@@ -1063,4 +1011,437 @@ size_t Grep::search(Query const& query, size_t limit, Archive& archive, File& co
+            if (j_is_cancelled) {
+                j_is_cancelled = false;
+                continue;
+            } else if ('\\' == processed_search_string[j]) {
+                j_is_cancelled = true;
+            }
+            std::vector<QueryLogtype> possible_substr_types;
+            // Don't allow an isolated wildcard to be considered a variable
+            if (i == j && is_greedy_wildcard[j]) {
+                possible_substr_types.emplace_back('*', "*", false);
+            } else if (i == j && is_non_greedy_wildcard[j]) {
+                possible_substr_types.emplace_back('?', "?", false);
+            } else {
+                set<uint32_t> variable_types;
+
+                // If the substring is preceded or proceeded by a greedy wildcard then it's possible
+                // the substring could be extended to match a var, so the wildcards are added to the
+                // substring. If we don't consider this case we could miss combinations. Take for
+                // example "* ab*cd *", "ab*" and "*cd" may both match a has# style variable
+                // ("\w*\d+\w*"). If we decompose the string into either substrings "* " + "ab*" +
+                // "cd" + " *" or "* " + "ab" + "*cd" + " *", neither would capture the possibility
+                // of a logtype with the form "* <has#><has#> *", which is a valid possibility
+                // during compression. Note, non-greedy wildcards do not need to be considered, for
+                // example "* ab?cd *" can never match "* <has#><has#> *".
+                uint32_t substr_start = j;
+                uint32_t substr_end = i;
+                bool prev_star = j > 0 && is_greedy_wildcard[j - 1];
+                bool next_star
+                        = i < processed_search_string.back() - 1 && is_greedy_wildcard[i + 1];
+                if (prev_star) {
+                    substr_start--;
+                }
+                if (next_star) {
+                    substr_end++;
+                }
+
+                // If the substring contains a wildcard, we need to consider the case that it can
+                // simultaneously match multiple variables and static text, and we need a different
+                // approach to compare against the archive.
+                bool contains_wildcard = false;
+
+                // If the substring isn't surrounded by delimiters there is no reason to consider
+                // the case where it is a variable as CLP would not compress it as such. Note:
+                // we must consider that wildcards could potentially be delimiters, and that the
+                // start and end of a log are also treated as delimiters.
+                bool has_preceding_delimiter
+                        = j == 0 || is_greedy_wildcard[j] || is_non_greedy_wildcard[j - 1]
+                          || lexer.is_delimiter(processed_search_string[j - 1]);
+                bool has_proceeding_delimiter
```

Comment:
As discussed offline, this should be `is_greedy_wildcard[i + 1]` (although once you make `i` exclusive, this statement will be correct already).